cmds/core/mv/mv.go

```
@@ -88,33 +87,40 @@ func mv(files []string, todir bool) error {
 	return nil
 }
 
-// For testing purpose
-var logFatalf = log.Fatalf
-
-func move(files []string, todir bool) {
+func move(files []string, todir bool) error {
 	dest := files[len(files)-1]
 	if destdir, err := os.Lstat(dest); err == nil {
 		todir = destdir.IsDir()
 	}
 	if len(files) > 2 && !todir {
 		fmt.Printf("Not a directory: %s\n", dest)
-		osExit(1)
-		return
+		return errOsExit
 	}
 
 	if err := mv(files, todir); err != nil {
-		logFatalf("%v", err)
+		return err
 	}
+	return nil
 }
 
 func main() {
 	var todir bool
 	flag.Parse()
 
 	if flag.NArg() < 2 {
-		usage()
+		err := usage()
+		if err == errOsExit {
+			os.Exit(1)
+		}
```

usage does not need to return an error. If usage is triggered just os.Exit(1) afterwards regardless